scvi/model/base/_pyromixin.py

```diff
@@ -0,0 +1,415 @@
+                )
+            else:
+                samples_ = self._get_posterior_samples(
+                    args, kwargs, **sample_kwargs_obs_plate
+                )
+
+                samples = {
+                    k: np.array(
+                        [
+                            np.concatenate(
+                                [samples[k][j], samples_[k][j]],
+                                axis=obs_plate_dim,
+                            )
+                            for j in range(
+                                len(samples[k])
+                            )  # for each sample (in 0 dimension
+                        ]
+                    )
+                    for k in samples.keys()  # for each variable
+                }
+            i += 1
+
+        # sample global parameters
+        tensor_dict = next(iter(train_dl))
+        args, kwargs = self.module._get_fn_args_from_batch(tensor_dict)
+        args = [a.to(device) for a in args]
+        kwargs = {k: v.to(device) for k, v in kwargs.items()}
+        self.to_device(device)
+
+        global_samples = self._get_posterior_samples(args, kwargs, **sample_kwargs)
+        global_samples = {
+            k: v
+            for k, v in global_samples.items()
+            if k not in list(self.module.model.list_obs_plate_vars()["sites"].keys())
+        }
+
+        for k in global_samples.keys():
+            samples[k] = global_samples[k]
+
+        self.module.to(device)
+
+        return samples
+
+    def sample_posterior(
+        self,
+        num_samples: int = 1000,
+        return_sites: Optional[list] = None,
+        use_gpu: bool = False,
+        batch_size: int = 128,
+        sample_kwargs=None,
+        return_samples: bool = False,
+    ):
+        """
+        Generate samples from posterior distribution for each parameter
+        and compute mean, 5%/95% quantiles, standard deviation.
+
+        Parameters
+        ----------
+        num_samples
+            number of posterior samples to generate.
+        return_sites
+            get samples for pyro model variable, default is all variables, otherwise list variable names).
+        use_gpu
+            Use gpu?
+        sample_kwargs
+            dictionary with arguments to _get_posterior_samples (see below):
+            return_observed
+                return observed sites/variables?
+        return_samples
+            return samples in addition to sample mean, 5%/95% quantile and SD?
+
+        Returns
+        -------
+        Posterior distribution samples, a dictionary with elements as follows,
+         containing dictionaries of numpy arrays for each variable:
+            post_sample_means - mean of the distribution for each variable;
+            post_sample_q05 - 5% quantile;
+            post_sample_q95 - 95% quantile;
+            post_sample_sds - standard deviation
+            posterior_samples - samples for each variable as numpy arrays of shape `(n_samples, ...)` (Optional)
```

Comment:
not sure this is proper formatting. You can see the docs build as one of the PR checks; though, you need to add these classes to the docs.

Reply:
@adamgayoso could you please point where these classes should be added to docs? I don't seem to be able to find it.